### DISCUSSION TITLE
1.20.0 scraping github project updates works pretty well

### DISCUSSION THREAD

**Discussion by SpiroCx** [FreshRSS/FreshRSS#4825] [Show and tell]
I tried this on the weekend and it's looking good compared to other github updates notification systems.  For those projects using github's releases system (<url>/releases)  there are already feeds, but for those projects without, this uses the commit hash as the title, content and unique id.  eg not necessary for freshrss because https://github.com/FreshRSS/FreshRSS/releases already has a feed, but for everything else (eg ranger-fm), this works.  You only need to change the url, the other fields below stay the same.  I am testing this out for tracking smaller projects (like tmux and vim plugins).

I am interested in suggestions for improvements.

url:
https://github.com/ranger/ranger

feed title: 
//main/div/div[1]/div/div/strong/a

finding news items:
item title:
item content:
item unique ID:
//main/turbo-frame/div/div/div/div[3]/div[1]/div[2]/div[1]/div/div[2]/div[2]/a[1]

Item link:
//main/div/div[1]/div/div/strong/a/@href

item date:
//main/turbo-frame/div/div/div/div[3]/div[1]/div[2]/div[1]/div/div[2]/div[2]/a[2]/relative-time/@datetime

**Comment by Alkarex** [comment]
It looks like some fields are missing, aren't they? Like title, content...
I think it would be more robust by targetting some classes instead of a deep series of div/div/div
I only made a quick test and could not get your suggestion to work.
What news are you trying to extract from https://github.com/ranger/ranger for instance?
If you are interested in the commits, try the Atom feed https://github.com/ranger/ranger/commits/master.atom

**Reply by SpiroCx** [reply]
The title is there, it's just the commit hash, but your master.atom feed is much better.  I will definitely try switching to that.  Thanks!

ps the way I wrote it isn't clear.  finding news items, item title, item content, item unique ID all use: //main/turbo-frame/div/div/div/div[3]/div[1]/div[2]/div[1]/div/div[2]/div[2]/a[1]
